optuna_integration/botorch/botorch.py

```diff
@@ -57,6 +60,22 @@ def _get_sobol_qmc_normal_sampler(num_samples: int) -> SobolQMCNormalSampler:
         def _get_sobol_qmc_normal_sampler(num_samples: int) -> SobolQMCNormalSampler:
             return SobolQMCNormalSampler(torch.Size((num_samples,)))
 
+    if version.parse(botorch.version.version) < version.parse("0.9.0"):
+
+        def _create_objective_and_kwargs(
```

Comment:
This allows us to continue supporting the old interface for old version of `BoTorch`. I can see two ways forward here:
- Continue to support old versions, but raise a warning here suggesting that a user upgrades to the newer version
- Drop support for old versions, and then this logic would not be needed

Reply:
I think it's OK to drop the old versions of botorch, since `BotorchSampler` itself is still experimental feature.